components/style/properties/longhand/box.mako.rs

```diff
@@ -1926,6 +1926,201 @@ ${helpers.single_keyword("transform-style",
     }
 </%helpers:longhand>
 
+<%helpers:longhand name="contain" animatable="False" products="gecko"
+                   spec="https://drafts.csswg.org/css-contain/#contain-property">
+    use std::fmt;
+    use style_traits::ToCss;
+    use values::HasViewportPercentage;
+
+    no_viewport_percentage!(SpecifiedValue);
+
+    pub mod computed_value {
+        #[derive(Clone, Debug, PartialEq)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub enum ComputedContainment {
+            Size,
+            Layout,
+            Style,
+            Paint,
+        }
+
+        #[derive(Clone, Debug, PartialEq)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub struct T(pub Option<Vec<ComputedContainment>>);
```

Comment:
Thanks, that sounds reasonable to me! I hadn't seen `bitflags!` before, but it does seem like the right approach. 😅 I'll try converting to that.